crates/noirc_evaluator/src/ssa/acir_gen/internal_var.rs

```diff
@@ -38,9 +38,15 @@ impl InternalVar {
     pub(crate) fn set_id(&mut self, id: NodeId) {
         self.id = Some(id)
     }
+    pub(crate) fn get_id(&self) -> Option<NodeId> {
+        self.id
+    }
     pub(crate) fn cached_witness(&self) -> &Option<Witness> {
         &self.cached_witness
     }
+    pub(crate) fn set_witness(&mut self, w: Option<Witness>) {
```

Comment:
Note: I was assuming before that we could only set the internalVar once, so this method would break that invariant

Reply:
We need to set the witness to handle for constant values.
We can have several NodeIds for the same value because they have different types (e.g 1 for bool, field or u32). Then they are distincts InternalVar, but we still want one witness for the same value.